external/dockerfile_functions.sh

```diff
@@ -84,34 +84,50 @@ print_image_args() {
     local build=$7
     local base_docker_registry_dir="$8"
 
-    image_name="docker.io/library/eclipse-temurin"
-    tag=""
-    if [[ "${package}" == "jre" ]]; then
-        tag="${version}-jre"
-    else
-        tag="${version}-jdk"
-    fi
+    image_name="$(getTemurinImageName)"
+    tag="$(getTemurinImageTag "${version}" "${package}")"
     if [[ "${vm}" == "openj9" ]]; then
         if  [[ "${os}" == "ubuntu" ]]; then
-            image_name="docker.io/ibm-semeru-runtimes"
-            tag=open-${tag}
+            image_name="$(getOpenJ9ImageName)"
+            tag="$(getOpenJ9ImageTag "${version}" "${package}")"
         elif [[ "${os}" == *"ubi"* && "${test}" != *"criu"* ]]; then
+            if isExternalImageEnabled ; then
+              echo "openj9 ubi and custom EXTERNAL_AQA_IMAGE are not compatible"
+              exit 1
+            fi
             image_name="registry.access.redhat.com/$base_docker_registry_dir"
             tag="latest"
+            EXTERNAL_AQA_IMAGE="${image_name}:${tag}"
         else
+            if isExternalImageEnabled ; then
+              echo "openj9 ubi and custom EXTERNAL_AQA_IMAGE are not compatible"
+              exit 1
+            fi
             # os is ubi, and test is criu
             # temporarily all ubi based testing use internal base image
             image_name="$DOCKER_REGISTRY_URL/$base_docker_registry_dir"
             tag="latest"
+            EXTERNAL_AQA_IMAGE="${image_name}:${tag}"
```

Comment:
If EXTERNAL_AQA_IMAGE needs to be set? If yes, should move to line 114 for both temurin and openj9 ?

Reply:
To construct the EXTERNAL_AQA_IMAGE artificially and globaly, is quite a good idea, and I was playing with it a lot.  The reason why I have not included it, are non deterministic OSes.  Ubuntu was hardcoded, to be there for jdk-xy-temurin/openj9, because there is no record of ubuntu in name.  From my limited knowledge, this is the only case where it is actually missing. Eg redhat java images still have ubi in name, and so on.

If there will be more usage of EXTERNAL_AQA_IMAGE then current  baseurl, path, name, os variables, then I guess such a crossroad table would be necessary, but for now I decided to not implement it.